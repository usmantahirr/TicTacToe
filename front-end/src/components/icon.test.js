import React from "react";
import { shallow } from "enzyme";

import Icon from "./icon";

describe("Icon", () => {
  it("should handle click event", () => {
    const clickHandler = jest.fn();
    const component = shallow(<Icon onClick={clickHandler} />);

    component.simulate("click");
    expect(clickHandler).toHaveBeenCalled();
    expect(component).toMatchSnapshot();
  });

  it("Should render children", () => {
    const component = shallow(
      <Icon>
        <i>sample icon</i>
      </Icon>
    );

    expect(component.children()).toHaveLength(1);
    expect(component.find("i").text()).toEqual("sample icon");
  });
});
