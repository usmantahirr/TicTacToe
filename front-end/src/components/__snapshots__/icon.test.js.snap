// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon should handle click event 1`] = `
<styled.button
  onClick={
    [MockFunction] {
      "calls": Array [
        Array [],
      ],
      "results": Array [
        Object {
          "type": "return",
          "value": undefined,
        },
      ],
    }
  }
/>
`;
